File: 2cfg.rb
#!/usr/bin/env ruby
require 'yaml'

# ==============================================================================
# OBJECT FUNCTIONS
# ==============================================================================
class Object
	def to_cfg(parent = '')
		return ["#{parent}=#{self.to_s}"]
	end
end

class Integer
	def to_cfg(parent = '')
		return ["#{parent}=#{self.to_s}"]
	end
end

class String
	def to_cfg(parent = '')
		return ["#{parent}='#{self}'"]
	end
end

class Array
	def to_cfg(parent = '')
		config = Array.new
		self.each_with_index do |item, index|
			config << [item.to_cfg("#{parent}[#{index}]")]
		end
		return config
	end
end

class Hash
	def to_cfg(parent = '')
		config = Array.new
		parent = "#{parent}." unless parent.empty?
		self.keys.sort.each do |key|
			config << self[key].to_cfg("#{parent}#{key}")
		end
		return config
	end
end


# ==============================================================================
# RUN
# ==============================================================================
puts YAML.load(ARGF).to_cfg

File: 2cfg.py
#!/usr/bin/env python3
import argparse
import sys
import yaml

parser = argparse.ArgumentParser()
parser.add_argument('input', nargs='?', type=argparse.FileType(), default=sys.stdin)
args = parser.parse_args()

info = yaml.safe_load(args.input)
#print(yaml.dump(info, indent=2))

def to_cfg_none(info, parent=''):
    return [f"{parent}=null"]

def to_cfg_int(info, parent=''):
    return [f"{parent}={info}"]

def to_cfg_str(info, parent=''):
    return [f"{parent}='{info}'"]

def to_cfg_list(info, parent=''):
    config = []
    for index, value in enumerate(info):
        config += to_cfg(value, f'{parent}[{index}]')
    return config

def to_cfg_dict(info, parent=''):
    if parent: parent = f'{parent}.'
    config = []
    for key, value in sorted(info.items()):
        config += to_cfg(value, f'{parent}{key}')
    return config

def to_cfg(info, parent = ''):
    if(info == None):
        return to_cfg_str(info, parent)
    elif(type(info) == int):
        return to_cfg_int(info, parent)
    elif(type(info) == str):
        return to_cfg_str(info, parent)
    elif(type(info) == list):
        return to_cfg_list(info, parent)
    elif(type(info) == dict):
        return to_cfg_dict(info, parent)
    else:
        raise TypeError(f"Unable to handle {type(info)}")

print("\n".join(to_cfg(info)))

